### DISCUSSION TITLE
Spatially Varying Viscosity

### DISCUSSION THREAD

**Discussion by hwhitehead** [PrincetonUniversity/athena#488] [Q&A] [answered]
From what I can tell, the answer to this question is "no", but I wil check all the same:

Is it possible to have a viscosity that is not constant throughout the simulation? Specifically I would like to vary the kinematic viscosity spatially and potentially temporarily.

**Comment by tomidakn** [comment, marked as answer]
It is not well documented yet, but you can use Mesh::EnrollViscosityCoefficient to enroll a function to set arbitrary viscosity nu(DiffProcess::iso,k,j,i).
https://github.com/PrincetonUniversity/athena/wiki/Diffusion-Processes
For details, please read the source code, hydro_diffusion.cpp.

**Reply by hwhitehead** [reply]
Thanks for the speedy response, I'll see what I can do with it. Cheers!
